src/main/kotlin/uk/gov/gdx/datashare/services/SuppliersService.kt

```diff
@@ -124,4 +124,27 @@ class SuppliersService(
       )
     }
   }
+
+  fun deleteSupplier(
+    id: UUID,
+  ): Supplier {
+    val now = dateTimeHandler.now()
+    adminActionAlertsService.noticeAction(
+      AdminAction(
+        "Delete supplier",
+        object {
+          val supplierId = id
+          val whenDeleted = now
+        },
+      ),
+    )
+    val supplier = supplierRepository.save(
+      supplierRepository.findByIdOrNull(id)?.copy(
```

Comment:
Should we just make `whenDeleted` mutable?

Reply:
If we're super keen on this being immutable, how would you feel about a `UPDATE ... RETURNING` repository method?